src/runner.ts

```diff
@@ -0,0 +1,157 @@
+        const coverageNode = this.getCoverageNode();
+
+        await this.generateCoverageSummary(coverageNode);
+        return { exitCode: 0 };
+    }
+
+    private async generateCoverageSummary(coverageNode: types.CoberturaCoverage | null) {
+        if (!coverageNode) {
+            throw new Error(messages.invalid_coverage_data);
+        }
+
+        const markdown = this.generateMarkdownContent(coverageNode.packages);
+        const totalCoverage = this.formatCoverage(coverageNode.linesCovered, coverageNode.linesValid, coverageNode.lineRate);
+
+        await core.summary
+            .addHeading('Parasoft Coverage')
+            .addRaw("<table><tbody><tr><th>Coverage&emsp;(covered/total - percentage)</th></tr>"
+                + "<tr><td><b>Total coverage&emsp;(" + totalCoverage + ")</b></td></tr>"
+                + markdown + "</tbody></table>")
+            .write();
+    }
+
+    private generateMarkdownContent(packagesNode: Map<string, types.CoberturaPackage>) {
+        if (!packagesNode || packagesNode.size === 0) {
+            throw new Error(messages.invalid_coverage_data);
+        }
+
+        const markdownRows: string[] = [];
+        for (const [packageName, packageNode] of packagesNode.entries()) {
+            const { coveredLines, totalLines, markdownContent } = this.calculatePackageCoverage(packageNode);
+            const packageCoverage = this.formatCoverage(coveredLines, totalLines, packageNode.lineRate);
+
+            markdownRows.push("<tr><td><details>" +
+                "<summary>" + packageName + "&emsp;(" + packageCoverage + ")</summary>" +
+                "<table><tbody>" + markdownContent + "</tbody></table>" +
+                "</details></td></tr>");
+        }
+
+        return markdownRows.join('');
+    }
+
+    private calculatePackageCoverage(packageNode: types.CoberturaPackage): { coveredLines: number, totalLines: number, markdownContent: string } {
+        if (!packageNode) {
+            throw new Error(messages.invalid_coverage_data);
+        }
+
+        let coveredLines = 0;
+        let totalLines = 0;
+        const markdownRows: string[] = [];
+
+        for (const classNode of packageNode.classes.values()) {
+            coveredLines += classNode.coveredLines;
+            totalLines += classNode.lines.length;
+            const classCoverage = this.formatCoverage(classNode.coveredLines, classNode.lines.length, classNode.lineRate);
+
+            markdownRows.push(`<tr><td>&emsp;${classNode.name}&emsp;(${classCoverage})</td></tr>`);
+        }
+
+        return { coveredLines, totalLines, markdownContent: markdownRows.join('') };
+    }
+
+    private formatCoverage(covered: number, total: number, rate: number): string {
+        if ((covered < 0 || total < 0)
+            || (covered > total)
+            || (rate < 0 || rate > 1)) {
+            throw new Error(messages.invalid_coverage_data);
+        }
+
+        return `${covered}/${total} - ${(rate * 100).toFixed(2)}%`;
```

Comment:
What if `total` equals to 0?

Reply:
Will add into condition if need. Actually total here is just used to display the result not as a dividend. Or, for it, we can return a specific message like:
if (total === 0) {
  return `${covered}/0 - N/A`; // or any other appropriate message
}

  return `${covered}/${total} - ${(rate * 100).toFixed(2)}%`;